lib/rspec-puppet/cache.rb

```diff
@@ -23,7 +23,7 @@ def get(*args, &blk)
     private
 
     def expire!
-      expired = @lra.slice!(0, @lra.size - MAX_ENTRIES)
+      expired = @lra.slice!(0, @lra.size - (MAX_ENTRIES-1))
```

Comment:
The -1 only helps with the case where the same args are used twice in a row at the border of MAX_ENTRIES, it does not help with the case when @ira[0] should be retained because the about to be added args are the same as @ira[0].
If args are give to expire! it could remove all those entries from @irl. In the case when the same args appear repeatedly all entries in the @ira will be the same, and there is nothing that should be evicted !